### PR TITLE
Multiple API improvements #627 #626 #119 #637

### PR DESCRIPTION
**Pull request by keith-turner**

### CODE REVIEW

File: modules/api/src/main/java/io/fluo/api/data/Bytes.java [hunk 1]
```diff
@@ -169,6 +169,9 @@ public final int hashCode() {
    */
   public static final Bytes of(byte[] array) {
     Objects.requireNonNull(array);
+    if (array.length == 0) {
```
Comment (mikewalch) [inline modules/api/src/main/java/io/fluo/api/data/Bytes.java]: Nice.  I like this change

**Comment by mikewalch** [comment]
Should `RowColumn` and `Column` have getter methods for row, cf, cq, etc that return `String`?

**Comment by keith-turner** [comment]
> Should RowColumn and Column have getter methods for row, cf, cq, etc that return String?

I was thinking about that, but was not sure what to name them.   The name would need be shorter than `getFamily().toString()`.

**Comment by mikewalch** [comment]
+1

**Comment by keith-turner** [comment]
I added String getters for data classes.  Following naming convention of snapshot.

**Comment by mikewalch** [comment]
String getters look good to me.

**Pull request merged**